confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,701 @@
+        cert = conf_copy.pop('ssl.certificate.location', None)
+
+        # If Tuple, (‘cert’, ‘key’) pair.
+        if cert is not None and key is not None:
+            self.session.cert = (cert, key)
+
+        # if String, path to ssl client cert file (.pem)
+        if cert is not None and key is None:
+            self.session.cert = cert
+
+        if key is not None and cert is None:
+            raise ValueError("ssl.certificate.location required when"
+                             " configuring ssl.key.location")
+
+        userinfo = utils.get_auth_from_url(base_url)
+        if 'basic.auth.user.info' in conf_copy:
+            if userinfo != ('', ''):
+                raise ValueError("basic.auth.user.info configured with"
+                                 " userinfo credentials in the URL."
+                                 " Remove userinfo credentials from the url or"
+                                 " remove basic.auth.user.info from the"
+                                 " configuration.")
+
+            userinfo = conf_copy.pop('basic.auth.user.info', '').split(':')
+
+            if len(userinfo) != 2:
+                raise ValueError("basic.auth.user.info must be in the form"
+                                 " of {username}:{password}")
+
+        # Auth handler or (user, pass) tuple. Defaults to Basic handler
+        self.session.auth = userinfo
+
+        # Any leftover keys are unknown to _RestClient
+        if len(conf_copy) > 0:
+            raise ValueError("Unrecognized property(ies) {}".format(conf_copy.keys()))
+
+    @property
+    def url(self):
+        return self.base_url
+
+    @property
+    def _auth(self):
+        return tuple(self.session.auth)
+
+    @property
+    def _certificate(self):
+        return self.session.cert
+
+    def close(self):
+        self.session.close()
+
+    def get(self, url, query=None):
+        return self.send_request(url, method='GET', query=query)
+
+    def post(self, url, body, **kwargs):
+        return self.send_request(url, method='POST', body=body)
+
+    def delete(self, url):
+        return self.send_request(url, method='DELETE')
+
+    def put(self, url, body=None):
+        return self.send_request(url, method='PUT', body=body)
+
+    def send_request(self, url, method, body=None, query=None):
+        """
+        Sends HTTP request to the SchemaRegistry.
+
+        All unsuccessful attempts will raise a SchemaRegistryError with the
+        response contents. In most cases this will be accompanied with a
+        Schema Registry supplied error code.
+
+        In the event the response is malformed an error_code of -1 will be used.
+
+        Args:
+            url (str): Request path
+            method (str): HTTP method
+            body (str): Request content
+            query (dict): Query params to attach to the URL
+
+        Returns:
+            dict: Schema Registry response content.
+        """
+        _headers = self.default_headers
```

Comment:
found the underscore prefix misleading, since it suggests this is a private member variable but this is a local variable.